File: src/BrewPuck/ClientApp/src/services/NhlApiService.ts
import HttpService, { IHttpService } from '@/services/HttpService';

const controller = "";
const basePath = "https://statsapi.web.nhl.com/api/v1/";

class NhlApiService {
    private readonly _http: IHttpService;

    constructor() {
        this._http = new HttpService(controller, false, basePath);
    }

    public async getSchedule(date?: string): Promise<Schedule> {
        if (date) {
            return this._http.getWithParams("schedule", { startDate: date, endDate: date, expand: "schedule.scoringplays" });
        } else {
            return this._http.get("schedule");
        }
    }

    public async getPlayer(id: number): Promise<{ people: Array<Player> }> {
        return this._http.get("people", id.toString());
    }

    public async getGameData(gamePk: number): Promise<LiveGame> {
        return this._http.get(`game/${gamePk}/feed/live?cb=${Date.now()}`);
    }

    public async getGamePatch(gamePk: number, startTimecode: string): Promise<Array<{ diff: Array<PatchOperation> }> | LiveGame>  {
        return this._http.get(`game/${gamePk}/feed/live/diffPatch?startTimecode=${startTimecode}&cb=${Date.now()}`);
    }

    public async getPlayerSeasonStats(playerId: number): Promise<PlayerSeasonStats> {
        const response: PlayerStatsResponse = await this._http.get(`people/${playerId}/stats?stats=statsSingleSeason`);
        try {
            return response.stats[0].splits[0].stat;
        } catch {
            return { goals: 0, points: 0, assists: 0, games: 0 } as PlayerSeasonStats;
        }
    }
}

export default new NhlApiService();

File: src/DraftPuck/Services/NhlApiService.cs
using DraftPuck.Models.NhlApi;
using Newtonsoft.Json.Linq;

namespace DraftPuck.Services
{
    public class NhlApiService : INhlApiService
    {
        private readonly HttpClient _httpClient;
        private readonly ILogger<NhlApiService> _logger;

        public NhlApiService(HttpClient httpClient, ILogger<NhlApiService> logger)
        {
            _httpClient = httpClient;
            _logger = logger;
        }

        public async Task<Schedule> GetScheduleAsync(bool withScoringPlays = false)
        {
            var path = "schedule";
            if (withScoringPlays) path += "?expand=schedule.scoringplays";
            return await _httpClient.GetFromJsonAsync<Schedule>(path);
        }

        public async Task<Schedule> GetScheduleAsync(DateTime date, bool withScoringPlays = false)
        {
            var path = $"schedule?startDate={date:yyyy-MM-dd}&endDate={date:yyyy-MM-dd}";
            if (withScoringPlays) path += "&expand=schedule.scoringplays";
            return await _httpClient.GetFromJsonAsync<Schedule>(path);
        }

        public async Task<Player> GetPlayerAsync(long playerId)
            => await _httpClient.GetFromJsonAsync<Player>($"people/{playerId}");

        public async Task<LiveGame> GetGameAsync(long gamePk)
            => await _httpClient.GetFromJsonAsync<LiveGame>($"game/{gamePk}/feed/live?cb={Guid.NewGuid()}");

        public async Task<JToken> GetPatchAsync(long gamePk, string startTimecode)
        {
            var str = await _httpClient.GetStringAsync($"game/{gamePk}/feed/live/diffPatch?startTimecode={startTimecode}&cb=${Guid.NewGuid()}");
            return JToken.Parse(str);
        }

        public async Task<PlayerStatsResponse> GetPlayerStatsAsync(long playerId)
            => await _httpClient.GetFromJsonAsync<PlayerStatsResponse>($"/people/{playerId}/stats?stats=statsSingleSeason");

        public async Task<LineScore> GetLinescoreForGameAsync(long gamePk)
            => await _httpClient.GetFromJsonAsync<LineScore>($"game/{gamePk}/linescore");
    }
}
